### ISSUE TITLE
Improve OpenApi docs by making extension methods more specific

### ISSUE THREAD

**Issue by DerStimmler** [co-IT/CSharpFunctionalExtensions.HttpResults#1]
Currently we make use of `JsonHttpResult` and `StatusCodeResult` to provide flexibililty without having to implement too many extension methods.

But that comes with a drawback regarding OpenApi docs.

E.g. you want to return an object with a Ok status code (200). With the current api you can call `.ToHttpResult()` on a `Result<T>` to return a `JsonHttpResult` with a status code of 200. You could also return another status code by providing it as a parameter like `.ToHttpResult(206)`.

The problem is that `JsonHttpResult` and `StatusCodeResult` don't implement `IEndpointMetadataProvider`. Therefore they aren't picked up correctly by the OpenApi generator.

The alternative would be to create specific extension methods for every `TypedResults.*` type.
So when you want to return the ok response you would have to call `.ToOkHttpResult()` and you cant modify the success status code. It's always 200 since we use `TypedResults.Ok<T>` in the background.

That means that the `.ToNoContentHttpResult()` will no longer return a `StatusCodeResult` where you can specify the status code, but will return a `NoContent` with fixed 204 status code.

---

In conclusion my suggestion is that we:
- rename `.ToHttpResult<T>()` to `.ToJsonHttpResult()`
- rename `.ToHttpResult()` to `.ToStatusCodeHttpResult()`
- change `.ToNoContentHttpResult()` to return `NoContent`
- add a `.ToStatusCodeHttpResult()` method that returns `StatusCodeHttpResult` for `Result<T>`
- add a `.ToOkHttpResult()` method that returns `Ok`
- add other extension methods for `TypedResults.*` in case of success results

Therefore we still have the flexibility and you can use `JsonHttpResult` and `StatusCodeResult` as you like. But you can also use the `Ok` and `NoContent` variant with specific types and OpenApi compatibility.

We don't need to adress failure results / error HttpResults since they can be customized via a custom error type and source generation.
